src/mlpack/methods/ann/loss_functions/log_cosh_loss.hpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file log_cosh_loss.hpp
+ * @author Kartik Dutt
+ *
+ * Definition of the Log-Hyperbolic-Cosine loss function.
+ *
+ * For more information, see the following paper.
+ * 
+ * @code
+ * article{
+ *   title   = {Log Hyperbolic Cosine Loss Improves Variational Auto-Encoder},
+ *   journal = {International Conference on Learning Representations},
+ *   year    = {2019}
+ * }
+ * @endcode
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_LOG_COSH_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_LOG_COSH_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Log-Hyperbolic-Cosine loss function is often used for to improve
+ * variational auto encoder. The Log-Hyperbolic-Cosine loss function
+ * taken log of hyperbolic cosine of difference between true values and
+ * predicted values.
```

Comment:
Minor readability suggestions: 
- replace "used for to" with "used to".
- Having "The Log-Hyperbolic-Cosine loss function" start two consecutive sentences doesn't read well; try a pronoun like "it" or "this function"
- instead of "taken" try "is the"

I'll do a more full readability lookover later.

Reply:
Hi @birm, I will make the changes. Thanks for pointing them out.

Reply:
Hi, I have made the changes. Thanks.